Flaky unit test TestExpectedPodDeletions_DeletionsSatisfied

The unit test `TestExpectedPodDeletions_DeletionsSatisfied` is flaky since this morning.

```go
=== RUN   TestExpectedPodDeletions_DeletionsSatisfied/two_deletions_expected,_only_one_satisfied
    --- FAIL: TestExpectedPodDeletions_DeletionsSatisfied/two_deletions_expected,_only_one_satisfied (0.00s)
        deletions_test.go:100: 
            	Error Trace:	deletions_test.go:100
            	Error:      	Not equal: 
            	            	expected: map[types.NamespacedName]types.UID{types.NamespacedName{Namespace:"ns", Name:"pod2"}:"41ca8fb2-718a-44c1-b26f-862e303baedd"}
            	            	actual  : map[types.NamespacedName]types.UID{types.NamespacedName{Namespace:"ns", Name:"pod1"}:"17578556-9e86-42ba-b1bd-50a23bffb356", types.NamespacedName{Namespace:"ns", Name:"pod2"}:"41ca8fb2-718a-44c1-b26f-862e303baedd"}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,2 +1,3 @@
            	            	-(map[types.NamespacedName]types.UID) (len=1) {
            	            	+(map[types.NamespacedName]types.UID) (len=2) {
            	            	+ (types.NamespacedName) ns/pod1: (types.UID) (len=36) "17578556-9e86-42ba-b1bd-50a23bffb356",
            	            	  (types.NamespacedName) ns/pod2: (types.UID) (len=36) "41ca8fb2-718a-44c1-b26f-862e303baedd"
            	Test:       	TestExpectedPodDeletions_DeletionsSatisfied/two_deletions_expected,_only_one_satisfied


Expected :map[types.NamespacedName]types.UID{types.NamespacedName{Namespace:"ns", Name:"pod2"}:"41ca8fb2-718a-44c1-b26f-862e303baedd"}
Actual   :map[types.NamespacedName]types.UID{types.NamespacedName{Namespace:"ns", Name:"pod1"}:"17578556-9e86-42ba-b1bd-50a23bffb356", types.NamespacedName{Namespace:"ns", Name:"pod2"}:"41ca8fb2-718a-44c1-b26f-862e303baedd"}
```
Full logs of two failed builds: 
https://devops-ci.elastic.co/job/cloud-on-k8s-pr/1915/consoleText
https://devops-ci.elastic.co/job/cloud-on-k8s-pr/1916/consoleText

--- a/pkg/controller/common/expectations/deletions.go
+++ b/pkg/controller/common/expectations/deletions.go
@@ -38,19 +38,20 @@ func (e *ExpectedPodDeletions) ExpectDeletion(pod corev1.Pod) {
 // the corresponding Pods do not exist in the cache anymore.
 // Expectations are cleared once fulfilled.
 func (e *ExpectedPodDeletions) DeletionsSatisfied() (bool, error) {
+	allSatisfied := true
 	for pod, uid := range e.podDeletions {
 		isDeleted, err := podDeleted(e.client, pod, uid)
 		if err != nil {
 			return false, err
 		}
 		if isDeleted {
-			// expectation fulfilled
+			// cache is up-to-date: expectation is fulfilled, remove it
 			delete(e.podDeletions, pod)
 		} else {
-			return false, nil
+			allSatisfied = false
 		}
 	}
-	return len(e.podDeletions) == 0, nil
+	return allSatisfied, nil
 }
 
 // podDeleted returns true if the pod with the given UID does not exist anymore.

